clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java

```diff
@@ -212,6 +215,36 @@ public boolean isVersionSupported(short apiVersion) {
         return apiVersion >= oldestVersion() && apiVersion <= latestVersion();
     }
 
+    public boolean isVersionEnabled(short apiVersion, boolean enableUnstableLastVersion) {
+        // ApiVersions API is a particular case that we always accept any, even
+        // unsupported, versions.
+        if (this == ApiKeys.API_VERSIONS) return true;
+
+        if (!messageType.latestVersionUnstable() || enableUnstableLastVersion) {
```

Comment:
Would it make sense to move this logic into some generated helpers? For example, `latestStableVersion` or `latestVersion(boolean enableUnstableVersions)`.

Reply:
Added `highestStableVersion(enableUnstableLastVersion)` to `ApiMessageType`.